modules/ROOT/pages/api-extensions.adoc

```diff
@@ -0,0 +1,429 @@
+= Anypoint DataGraph API Extension Reference
+
+This guide defines API extensions for DataGraph. 
+
+
+== Object Type Extensions 
+
+This extension marks that an object type has a primary key, and thus can be used as a reference or extension.
+
+This extension applies to `ObjectType`.
+
+=== Structure
+
+[%header%autowidth.spread]
+|===
+|Field |Type |Description
+|`entity` |Map |Marks an object type as an entity.
+|`entity.key` |String |An array that must not be empty. Each item must reference a field that is a key of the entity.
+|===
+
+=== OAS Example
+
+----
+components:
+  schemas:
+    Customer:
+  	x-entity:
+    	  key: [id]
+  	properties:
+    	  name:
+          type: string
+    	  email:
+          type: string
+    	  id:
+          type: string
+    	  ...
+----
+
+=== RAML Example
+
+----
+types:
+  Customer:
+    (dg.entity):
+      key: [id]
+    properties:
+    	name:
+        type: string
+    	email:
+        type: string
+    	id:
+        type: string
+    	...
+----
+
+== Entity Reference
```

Comment:
Our entity reference extension its used with "entity-references" instead of "entity-reference".
We had a typo in our document, and we will update it soon too.

Reply:
got it